core/src/main/java/org/apache/calcite/adapter/enumerable/ReflectiveCallNotNullImplementor.java

```diff
@@ -47,17 +47,34 @@ public Expression implement(RexToLixTranslator translator,
       RexCall call, List<Expression> translatedOperands) {
     translatedOperands =
         EnumUtils.fromInternal(method.getParameterTypes(), translatedOperands);
+    final Expression callExpr;
     if ((method.getModifiers() & Modifier.STATIC) != 0) {
-      return Expressions.call(method, translatedOperands);
+      callExpr = Expressions.call(method, translatedOperands);
     } else {
       // The UDF class must have a public zero-args constructor.
       // Assume that the validator checked already.
       final Expression target =
           Expressions.new_(method.getDeclaringClass());
-      return Expressions.call(target, method, translatedOperands);
+      callExpr = Expressions.call(target, method, translatedOperands);
     }
+    if (!containsCheckedException(method)) {
+      return callExpr;
+    }
+    return translator.handleMethodCheckedExceptions(callExpr);
   }
 
+  private boolean containsCheckedException(Method method) {
+    Class[] exceptions = method.getExceptionTypes();
+    if (exceptions == null || exceptions.length == 0) {
+      return false;
+    }
+    for (Class clazz: exceptions) {
```

Comment:
Sorry this is getting really nitpicky, but there should be a space on either side of the `:`.